### PR TITLE
New slugify function

### PR DESCRIPTION
**Pull request by GreatWizard**
- [x] Clean code
- [x] Change camelize by  dasherize

review: @xcambar

### CODE REVIEW

File: tests/unit/utils/string-test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+test('tests on removeDiacritics', (assert)=> {
+  assert.equal(removeDiacritics(''), '');
+  assert.equal(removeDiacritics('é è à ù ç č Ž ñ'), 'e e a u c c Z n');
+});
+
+test('tests on slugify', (assert)=> {
+  assert.equal(slugify(''), '');
+  assert.equal(slugify('bonjour'), 'bonjour');
+  assert.equal(slugify('annyǒng hashimnikka'), 'annyong-hashimnikka');
+  assert.equal(slugify('Bon jOuR Töi 1337   '), 'bon-j-ou-r-toi-1337');
```
Comment (xcambar) [inline tests/unit/utils/string-test.js]: Isn't `bon-jou-r-toi-1337` expected instead?

I don't understand the `j-o`

**Comment by xcambar** [comment]
👎

**Comment by xcambar** [comment]
Please see my comments on the slugify function

File: tests/unit/utils/string-test.js (hunk 1 above)
Comment (GreatWizard) [inline tests/unit/utils/string-test.js]: capitalized character is a word breaker

Comment (GreatWizard) [inline tests/unit/utils/string-test.js]: http://emberjs.com/api/classes/Ember.String.html#method_dasherize

**Comment by xcambar** [comment]
👍

**Pull request merged**